ARE/services/WebService/src/main/java/eu/asterics/mw/webservice/serverUtils/ServerRepository.java

```diff
@@ -35,81 +39,152 @@
  *
  */
 public class ServerRepository {
+    // Singleton instance
+    private static ServerRepository instance;
 
     // Rest Server configuration
+    public static final String ARE_WEBSERVICE_PORT_REST_KEY = "ARE.webservice.port.REST";
     public static final String PATH_REST = "/rest";
-    public static final int PORT_REST = 8081;
-    public static final URI BASE_URI_REST = URI.create("http://0.0.0.0:" + PORT_REST + PATH_REST);
-
-    // Web Service Server configuration
-    public static final String PATH_WS = "/ws";
-    public static final String PATH_WS_ASTERICS_DATA = "/astericsData";
-    public static final int PORT_WS = 8082;
-    public static final URI BASE_URI_WS = URI.create("http://localhost:" + PORT_WS + PATH_WS);
+    public static final int DEFAULT_PORT_REST = 8081;
+
+    // Web Socket Server configuration
+    public static final String PATH_WEBSOCKET = "/ws";
+    public static final String PATH_WEBSOCKET_ASTERICS_DATA = "/astericsData";
+    public static final String ARE_WEBSERVICE_PORT_WEBSOCKET_KEY = "ARE.webservice.port.websocket";
+    public static final int DEFAULT_PORT_WEBSOCKET = 8082;
+
+    // member variables holding property values
+    private int portREST = DEFAULT_PORT_REST;
+    private int portWebsocket = DEFAULT_PORT_WEBSOCKET;
+
+    /**
+     * Private ctor used for initializing the class.
+     */
+    private ServerRepository() {
+        // init ports and paths with property values
+        try {
+            portREST = Integer.parseInt(AREProperties.instance.getProperty(ARE_WEBSERVICE_PORT_REST_KEY, String.valueOf(DEFAULT_PORT_REST)));
+        } catch (NumberFormatException e) {
+            AstericsErrorHandling.instance.getLogger().logp(Level.WARNING, this.getClass().getName(), "ServerRepository()",
+                    "Configured port for REST service invalid: " + e.getMessage(), e);
+        }
+        // init ports and paths with property values
+        try {
+            portWebsocket = Integer.parseInt(AREProperties.instance.getProperty(ARE_WEBSERVICE_PORT_WEBSOCKET_KEY, String.valueOf(DEFAULT_PORT_WEBSOCKET)));
+        } catch (NumberFormatException e) {
+            AstericsErrorHandling.instance.getLogger().logp(Level.WARNING, this.getClass().getName(), "ServerRepository()",
+                    "Configured port for Websocket service invalid: " + e.getMessage(), e);
+        }
+    }
+
+    /**
+     * Returns the baseURI for the REST API.
+     * 
+     * @return the baseUriRest
+     */
+    public URI getBaseUriREST() {
+        return URI.create("http://0.0.0.0:" + getPortREST() + PATH_REST);
+    }
+
+    /**
+     * Returns the baseURI for the websocket functionality. The actual websocket channels must be subpaths of it, e.g.
+     * {@link ServerRepository#PATH_WEBSOCKET_ASTERICS_DATA}
+     * 
+     * @return the baseUriWs
+     */
+    public URI getBaseUriWebsocket() {
+        return URI.create("http://localhost:" + getPortWebsocket() + PATH_WEBSOCKET);
```

Comment:
maybe this line should also be changed to 0.0.0.0, because the actual webservice is initialized with 0.0.0.0 in WebServiceEngine:109. However it is not so important, so I'll merge this now.

Reply:
interesting point, should be tested by times with 0.0.0.0

Reply:
it actually uses 0.0.0.0 -> see WebServiceEngine:109. this method is only used for logging

Reply:
so if the address string was configurable in areProperties or maybe event through the REST API, the restriction of remote access to localhost could be achieved easily, right?
See https://github.com/asterics/AsTeRICS/issues/87

Reply:
i do not know the details of grizzly webserver, but if is correct that with initializing the address to localhost does not allow external access, thats correct.

however, i would define a property like:
`webserver.enable.external.access=true/false`
and internally set the address to "0.0.0.0" or "localhost" - thats much more intuitive/understandable.